src/excmds.ts

```diff
@@ -372,29 +373,21 @@ export function fillcmdline(...strarr: string[]) {
     message("commandline_frame", "fillcmdline", [str])
 }
 
-// TODO: For security, this should really be in the background.
-// Extensions to the macros and messaging systems required, however.
-//#content
-export function clipboard(excmd = "open"){
-    let scratchpad = document.createElement("textarea")
-    // Scratchpad must be `display`ed, but can be tiny and invisible.
-    // Being tiny and invisible means it won't make the parent page move.
-    scratchpad.style.cssText = 'visible: invisible; width: 0; height: 0; position: fixed'
-    scratchpad.contentEditable = "true"
-    document.documentElement.appendChild(scratchpad)
-    if (excmd == "yank"){
-        scratchpad.value = window.location.href
-        scratchpad.select()
-        document.execCommand("Copy")
-    } else if (excmd == "open"){
-        scratchpad.focus()
-        document.execCommand("Paste")
-        const url = scratchpad.textContent
-        if (url) open(url)
+//#background
+export async function clipboard(excmd = "open"){
+    switch (excmd) {
+        case 'yank':
+            const atab = await activeTab()
+            message("commandline_frame", "setclipboard", [atab.url])
+            break
+        case 'open':
+            message("commandline_frame", "gotourl", [])
+            break
+        default:
+            // todo: maybe we should have some common error and error handler
```

Comment:
Yes, definitely. I think it's in messaging.ts already, but excmds.ts uses the old messaging system. I'll fix it.